src/playercards/PlayerCardPanel.ttslua

```diff
@@ -755,3 +788,122 @@ function spawnRandomWeakness()
     rotation = FACE_UP_ROTATION,
   })
 end
+
+function spawnCardList(params)
+  local cardList      = params.cardList
+  local name          = params.name
+  local startPos      = params.startPos or "other"
+  local rotation      = params.rotation or FACE_UP_ROTATION
+  local spread        = params.spread or true
+  local spreadCols    = params.spreadCols or 20
+  local targetState   = params.targetState or "default"
+  prepareToPlaceCards(targetState)
+  local groupPos = Vector(startPositions[startPos])
+  if #cardList > 0 then
+    spawnBag.spawn({
+      name = name,
+      cards = cardList,
+      globalPos = self.positionToWorld(groupPos),
+      rotation = rotation,
+      spread = spread,
+      spreadCols = spreadCols
+    })
+  end
+end
+
+-- @param params table:
+--    cardList: 
+--    name:
+--    startPos:
+--    rotation:
+--    spread:
+--    spreadCols:
+function spawnCardsByType(params)
```

Comment:
This function looks very repetitive to me. Would it be possible to rework it for easier readability and maintainability?